### PR TITLE
truss chains push sends chainlet details to enable promotion to environment with proper settings

### PR DESCRIPTION
**Pull request by rcano-baseten**
## :rocket: What
* Correlary to https://github.com/basetenlabs/baseten/pull/9655 
* Needs to be merged after ^

## :microscope: Testing
* tested e2e locally

### CODE REVIEW

File: truss/remote/baseten/api.py [hunk 1]
```diff
@@ -109,6 +109,9 @@ def create_model_from_truss(
         is_trusted: bool,
         deployment_name: Optional[str] = None,
         origin: Optional[b10_types.ModelOrigin] = None,
+        chain_environment: Optional[str] = None,
```
Comment (spal1) [inline truss/remote/baseten/api.py:112]: we should make sure the django mutation to accept these inputs is merged (and makes it all the way to production) before merging this PR

**Review by spal1 — APPROVED** [review]

File: truss-chains/truss_chains/remote.py [hunk 2]
```diff
@@ -124,7 +127,7 @@ def _push_service(
         )
     elif isinstance(options, definitions.PushOptionsBaseten):
         with utils.log_level(logging.INFO):
-            service = _push_to_baseten(truss_dir, options)
+            service = _push_to_baseten(truss_dir, options, chainlet_descriptor.name)
```
Comment (spal1) [inline truss-chains/truss_chains/remote.py:130]: do we want the `name` or the `display_name` here?

Comment (spal1) [reply truss-chains/truss_chains/remote.py:130]: looks like we use `display_name` when we send the chainlet mapping over 
https://github.com/basetenlabs/truss/blob/main/truss-chains/truss_chains/remote.py#L316-L320

Comment (rcano-baseten) [reply truss-chains/truss_chains/remote.py:130]: great catch ty

**Comment by squidarth** [comment]
will shepherd the release of this tomorrow

**Review by squidarth — APPROVED** [review]

**Comment by spal1** [comment]
validated this against production -- going to merge

**Pull request merged**